packages/resolve-command/src/index.js

```diff
@@ -1,72 +1,83 @@
+        const eventStream = eventStore.getStreamByAggregateId(aggregateId);
 
-function executeCommand(command, aggregate, store) {
-    return getAggregateState(aggregate, command.aggregateId, store).then((aggregateState) => {
-        const handler = aggregate.commands[command.type];
-        const event = handler(aggregateState, command);
+        eventStream.on('readable', () => {
+            let event;
+            // eslint-disable-next-line no-cond-assign
+            while (null !== (event = eventStream.read())) {
+                const handler = handlers[event.type];
+                if (handler) {
+                    aggregateState = handler(aggregateState, event);
+                }
+            }
+        });
 
-        if (!event.type) {
-            return Promise.reject('event type is required');
-        }
+        eventStream.on('end', () => resolve(aggregateState));
 
-        event.aggregateId = command.aggregateId;
-        event.timestamp = Date.now();
-        return event;
+        eventStream.on('error', reject);
     });
 }
 
-function saveEvent(event, store) {
-    return store.saveEvent(event).then(() => event);
-}
+async function executeCommand(command, aggregate, eventStore) {
+    const aggregateState = await getAggregateState(aggregate, command.aggregateId, eventStore);
 
-function publishEvent(event, bus) {
-    bus.emitEvent(event);
+    const handler = aggregate.commands[command.type];
+    const event = handler(aggregateState, command);
+
+    if (!event.type) {
+        throw new Error('event type is required');
+    }
+
+    event.aggregateId = command.aggregateId;
+    event.timestamp = Date.now();
     return event;
 }
 
-function createAggregateExecutor({ store, bus, aggregate }) {
-    return command =>
-        executeCommand(command, aggregate, store)
-            .then(event => saveEvent(event, store))
-            .then(event => publishEvent(event, bus));
+function propagateEvent(event, publishStream) {
+    return new Promise((resolve, reject) =>
+        publishStream.write(event, err => (err ? reject(err) : resolve(event)))
+    );
+}
+
+function createExecutor({ eventStore, aggregate, publishStream }) {
+    return async (command) => {
+        const event = await executeCommand(command, aggregate, eventStore);
+        return await propagateEvent(event, publishStream);
+    };
 }
 
-function createExecutor({ store, bus, aggregates }) {
+export default ({ eventStore, aggregates }) => {
+    const publishStream = eventStore.getPublishStream();
+
     const executors = aggregates.reduce((result, aggregate) => {
-        result[aggregate.name.toLowerCase()] = createAggregateExecutor({
-            store,
-            bus,
-            aggregate
+        result[aggregate.name.toLowerCase()] = createExecutor({
+            eventStore,
+            aggregate,
+            publishStream
         });
         return result;
     }, {});
 
     return command =>
```

Comment:
No, upper function is pure synchronous, and it is factory of async functions (With closure on it's arguments). It is not same.